### DISCUSSION TITLE
ekg version 0.5 about to be released, testing and feedback is needed

### DISCUSSION THREAD

**Discussion by ahyatt** [ahyatt/ekg#135] [Announcements]
Hi all.  It's been a while since the last release, and we have some significant changes:

- Trash is now just a tag, and I've removed the old concept of trashed tags.  When you update to the latest commit in the `develop` branch, it *should* look at all notes with trashed tags, and if it's all trashed tags, it will just add the trash tag.  But if it's some trashed tags and some not, it will let you know, and you can use the new `ekg-show-notes-with-tag-prefix` with "trash/" to see all those notes.  Does this work well?  I'd like to know!  @qingshuizheng, this also fixes the issue you noted with inline tags interfering with trashed state.

- @qingshuizheng has made the auto-save extension, and functions to cancel captures and edits.  Does it make sense, and does it work like you think it should?

- Inline tags are here.  Do they work well?  Do they cause unintended side-effects?

- Org-mode commands should work better, at least the `org-insert` ones.  Are there others that should also have special treatment because they don't work well with the metadata section?

I'll plan on releasing later this week, perhaps on the weekend.  Feedback before then would be helpful, so thank you!

**Comment by qingshuizheng** [comment]
Hello @ahyatt ,

I running errands out of town, and am not able to run more tests.

The trash tag revamp looks good, and I like the idea.

For Auto-save and ekg--kill-buffer-query-functions, I think there's confusion between "abort capture" and "abort edit" at first glance (at least from your first response when this feature was brought in, but I might be wrong.) So I am considering to change the command options to:

1. Finish
2. Save (as draft)
3. Discard changes (for ekg-edit only)
4. Delete this note.

Would this options be more clearer than the current one?

I don't think I have time to do it in the following week to write you a PR. Also this options above might be too long for the header line.

Or we could sleep on it and decide if this change is necessary in the next next next release.

I'm typing on my phone, there's anything needing more explanation, let me know.

**Reply by ahyatt** [reply]
I like the idea, but let me take a look at it, perhaps implement it and see how it looks in practice.  Thanks, I also agree that this is a bit confusing at the moment.

**Reply by ahyatt** [reply]
After thinking about this for a while, I went a different way to simplify things as much as possible:

I removed the ability to abort edits, which I never really understood.  Also, I replaced the complicated choices when killing buffers to just to kill or not.  If the user wants to save, they can not kill, save, and the kill again, and so on for other choices.  I made a few other changes as well, please take a look and let me know if I've simplified it too much and removed functionality that you think is necessary.

**Comment by gour** [comment]
Hello,

just wonder what is with the FTS feature? Next release?

**Reply by ahyatt** [reply]
It's pretty close, but I noticed a bug where it wasn't giving me all the results I should have gotten, and I can't figure out why.  If not for that, it'd go out in this release.  But I intend to push it out soon after it is ready.  Thanks for asking!

**Reply by gour** [reply]
> It's pretty close, but I noticed a bug where it wasn't giving me all the results I should have gotten, and I can't figure out why. 

Does fts feature provide complete query syntax for tags as well?

**Reply by ahyatt** [reply]
> Does fts feature provide complete query syntax for tags as well?

Not as currently implemented - there's no predicate specific syntax, it's all just mixed in.  This is a consequence of the generic FTS implementation based on the triples package.  If we were to do a specific FTS implementation for ekg, that should make this possible, although it's more complicated.

**Comment by qingshuizheng** [comment]
I like your idea to make life easier. I may have overdone it.

Sorry that I am not able to test it on my computer at the moment, I’m still out of down and surrounded by some business. Will return in a few days. But I think you are good to make a release if things are running okay.
